lib/SILOptimizer/Mandatory/DIMemoryUseCollector.cpp

```diff
@@ -835,6 +850,11 @@ void ElementUseCollector::collectUses(SILValue Pointer, unsigned BaseEltNo) {
       continue;
     }
 
+    // Sanitizer instrumentation is not user visible, so it should not
+    // count as a use and must not affect compile-time diagnostics.
+    if (isSanitizerInstrumentation(User, Module.getASTContext()))
+      continue;
```

Comment:
@devincoughlin I would mess with that later TBH. Lets get this in as a first iteration.